src/pipeline/synth_data_gen.py

```diff
@@ -0,0 +1,176 @@
+import os
+import toml
+import json
+import asyncio
+import numpy as np
+from tqdm import tqdm
+from string import Template
+from typing import List, Dict
+from datasets import (
+    Dataset, DatasetDict, load_dataset
+)
+
+from ..gen.gemini import get_model as get_service_model
+from ..gen.utils import call_service_llm
+
+def _load_all_json_files(filenames):
+    """
+    _load_all_json_files loads up JSON files as Python dictionaries
+    """
+    all_json_dicts = []
+    for filename in filenames:
+        with open(filename) as f:
+            all_json_dicts.append(json.load(f))
+    return all_json_dicts
+
+def _format_response(responses: List[Dict[str, str]]):
+    """
+    _format_response extracts information from the Python dictionaries that
+    consists of synthetic data, and then reformat them as a standard message
+    format ([{'role': ... 'content': ...}]).
+    """
+    final_instruction_answer_pairs = []
+    seed_prompts = [responses["seed_prompt"]] * len(responses["contents"])
+
+    for response in responses["contents"]:
+        user_response_dict = {}
+        assistant_response_dict = {}
+        user_response_dict["content"] = response["instruction"]
+        user_response_dict["role"] = "user"
+        assistant_response_dict["content"] = response["response"]
+        assistant_response_dict["role"] = "assistant"
+
+        final_instruction_answer_pairs.append([user_response_dict, assistant_response_dict])
+
+    return seed_prompts, final_instruction_answer_pairs
+
+def _sampling(dataset_id, split, num_sample):
+    """
+    _sampling samples a nubmer of data (as many as specified num_sample)
+    from the given dataset_id[split]
+    """
+    samples = []
+    ds = load_dataset(dataset_id, split=split)
+    total_original_samples = len(ds)
+    random_indices = np.random.randint(
+        0, total_original_samples, size=(num_sample)
+    ).tolist()
```

Comment:
`tolist()` should cut it no? Why the extra loop?

Reply:
I was looking for a better way to filter out a list based on indicies. 

You did it as `EXISTING_DATASET[int(random_index)]`, and I wanted to use the same code, but I thought making `_sampling()` to return the exact samples not indicies is more comprehensive.

Reply:
Then we should probably use the `select()` method of `datasets`. I will let you consult the documentation.